smart_importer/predictor.py

```diff
@@ -89,13 +86,21 @@ def __call__(
         Returns:
             A list of entries, modified by this predictor.
         """
-        logging.debug("Running %s for file %s", self.__class__.__name__, file)
-        self.account = importer.account(file)
         self.load_training_data(existing_entries)
         with self.lock:
             self.define_pipeline()
             self.train_pipeline()
-            return self.process_entries(imported_entries)
+            result = []
+            for filename, entries, account, importer in imported_entries:
```

Comment:
how about making this compatible with being as a hook from Fava as well as from beangulp? (2-tuples / 4-tuples)